### PR TITLE
Mention reverse proxy for PREFECT_API_URL config

### PR DESCRIPTION
**Pull request by hozn**
It took me some trial & error to figure out that not only does an *agent* need to know about the `PREFECT_API_URL`, but the server itself needs to have this value configured when the server is running behind a reverse proxy.

### Example

In my particular example, I added the environment variable to my Prefect Orion server deployment (Nomad):
```
PREFECT_API_URL='https://prefect-orion-dev.mycompany.tld/api'
```

### Checklist
- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Comment by madkinsz** [comment]
Hi @hozn thanks for contributing!

We should probably mention the `PREFECT_ORION_UI_API_URL` instead (https://docs.prefect.io/api-ref/prefect/settings/#prefect.settings.PREFECT_ORION_UI_API_URL), it pulls its default value from the `PREFECT_API_URL` for convenience but this is the setting that is actually relevant here.

**Comment by hozn** [comment]
@madkinsz -- ah, thank you!  I wasn't aware of that one :)  Ok, I can look at fixing this (moving this to that section).

**Comment by hozn** [comment]
@madkinsz , I updated the wording, though I don't know if you had instead the idea that there would be a new configuration section for PREFECT_ORION_UI_API_URL (?)

**Review by madkinsz — APPROVED** [review]
Thanks!

File: docs/concepts/settings.md [hunk 1]
```diff
@@ -41,6 +41,11 @@ PREFECT_API_URL="https://api.prefect.cloud/api/accounts/[ACCOUNT-ID]/workspaces/
 
     If you want the agent to communicate with Prefect Cloud or a Prefect Orion API server instance from a remote execution environment such as a VM or Docker container, you must configure `PREFECT_API_URL` in that environment.
 
+
+!!! tip "`PREFECT_ORION_UI_API_URL` (and `PREFECT_API_URL`) setting when running Prefect Orion UI behind a reverse proxy"
```
Comment (tpdorsey) [inline docs/concepts/settings.md:45]:
```suggestion
!!! tip "Running Prefect Orion UI behind a reverse proxy"
```

File: docs/concepts/settings.md [hunk 2]
```diff
@@ -41,6 +41,11 @@ PREFECT_API_URL="https://api.prefect.cloud/api/accounts/[ACCOUNT-ID]/workspaces/
...
+    When using a reverse proxy (such as [Nginx](https://nginx.org) or [Traefik](https://traefik.io)) to proxy traffic to a locally-hosted Prefect Orion UI instance, the Prefect Orion server also needs to be configured to know how to connect to the API. The  [`PREFECT_ORION_UI_API_URL`](prefect/settings/#prefect.settings.PREFECT_ORION_UI_API_URL)  should be set to the external proxy URL (e.g. if your external URL is https://prefect-orion.example.com/ then set `PREFECT_ORION_UI_API_URL=https://prefect-orion.example.com/api` for the Prefect Orion server process).  You can also accomplish this by setting [`PREFECT_API_URL`](/concepts/settings/#prefect.settings.PREFECT_API_URL) to the API URL, as this setting is used as a fallback if `PREFECT_ORION_UI_API_URL` is not set.
```
Comment (tpdorsey) [inline docs/concepts/settings.md:46]:
```suggestion
    When using a reverse proxy (such as [Nginx](https://nginx.org) or [Traefik](https://traefik.io)) to proxy traffic to a locally-hosted Prefect Orion UI instance, the Prefect Orion server also needs to be configured to know how to connect to the API. [`PREFECT_ORION_UI_API_URL`](prefect/settings/#prefect.settings.PREFECT_ORION_UI_API_URL) should be set to the external proxy URL. For example, if your external URL is `https://prefect-orion.example.com/`, then set `PREFECT_ORION_UI_API_URL=https://prefect-orion.example.com/api` for the Prefect Orion server process).  You can also accomplish this by setting [`PREFECT_API_URL`](/concepts/settings/#prefect.settings.PREFECT_API_URL) to the API URL, as this setting is used as a fallback if `PREFECT_ORION_UI_API_URL` is not set.
```

**Review by tpdorsey — COMMENTED** [review]
Minor copyedits for consistency. This is great. Thank you.

File: docs/concepts/settings.md (hunk 2 above)
Comment (hozn) [reply docs/concepts/settings.md:46]: @tpdorsey , thank you!

**Review by madkinsz — APPROVED** [review]

**Pull request merged**